pkg/router/base_rule.go

```diff
@@ -18,17 +18,19 @@
 package router
 
 import (
+	"context"
 	"math/rand"
 	"strings"
 	"sync"
 	"time"
 
 	"mosn.io/api"
-	"mosn.io/mosn/pkg/config/v2"
+	v2 "mosn.io/mosn/pkg/config/v2"
 	"mosn.io/mosn/pkg/log"
 	"mosn.io/mosn/pkg/protocol"
 	httpmosn "mosn.io/mosn/pkg/protocol/http"
 	"mosn.io/mosn/pkg/types"
+	"mosn.io/mosn/pkg/upstream/cluster"
```

Comment:
caused import cycle https://travis-ci.com/github/mosn/mosn/jobs/347205595

Reply:
I have run command and nothing changed , what can i do for it ? 

Thank you!

```
goimports -w pkg/router/base_rule.go  
goimports -w pkg/router/base_rule_test.go
```

Reply:
> I have run command and nothing changed , what can i do for it ?
> 
> Thank you!
> 
> ```
> goimports -w pkg/router/base_rule.go  
> goimports -w pkg/router/base_rule_test.go
> ```

It is the package import cycle that causes the [CI task]( https://travis-ci.com/github/mosn/mosn/jobs/347205595) to fail, as  below

```
--------Run test package: mosn.io/mosn/pkg/upstream/cluster
# mosn.io/mosn/pkg/upstream/cluster
import cycle not allowed in test
package mosn.io/mosn/pkg/upstream/cluster (test)
	imports mosn.io/mosn/pkg/router
	imports mosn.io/mosn/pkg/upstream/cluster
FAIL	mosn.io/mosn/pkg/upstream/cluster [setup failed]
make: *** [Makefile:33: coverage-local] Error 1
make: *** [coverage] Error 2
The command "make coverage" exited with 2.
```

Reply:
I tried to solve this `import cycle not allowed in test` , but it not working.   

I am not very skilled of golang , do you have any suggestions ? 

Thank you for your help!

Reply:
It has been resolved.